streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1877,12 +1877,10 @@ public synchronized ClientInstanceIds clientInstanceIds(final Duration timeout)
         // (1) fan-out calls to threads
 
         // StreamThread for main/restore consumers and producer(s)
-        final Map<String, KafkaFuture<Uuid>> consumerFutures = new HashMap<>();
-        final Map<String, KafkaFuture<Map<String, KafkaFuture<Uuid>>>> producerFutures = new HashMap<>();
```

Comment:
Originally, we did split consumer and producer to be able to handle the producer-per-task case. Now, we only have producer-per-thread and thus can unify consumer/producer into "clients"